pkg/config/config.go

```diff
@@ -189,18 +189,18 @@ func validateConfig(cfg *Config) error {
 	} else {
 		b, err = hex.DecodeString(cfg.PrivateKey)
 		if err != nil || len(b) != 32 {
-			return errors.Errorf("expecting 64 hex character private key, got \"%s\"", cfg.PrivateKey)
+			return errors.Wrapf(err, "expecting 64 hex character private key, got \"%s\"", cfg.PrivateKey)
 		}
 		if len(cfg.ContractAddress) != 42 {
-			return errors.Errorf("expecting 40 hex character contract address, got \"%s\"", cfg.ContractAddress)
+			return errors.Wrapf(err, "expecting 40 hex character contract address, got \"%s\"", cfg.ContractAddress)
```

Comment:
```suggestion
			return errors.Errorf("expecting 40 hex character contract address, got \"%s\"", cfg.ContractAddress)
```
There is no error to wrap here. You are printing the error from the last if block

The same in few other places